### PR TITLE
Added D. mel model documentation

### PR DESCRIPTION
**Pull request by ndukler**
Documentation for the two d. mel. models. Also added the metadata for the populations.

### CODE REVIEW

File: stdpopsim/drosophila_melanogaster.py [hunk 1]
```diff
@@ -75,13 +75,44 @@ class Comeron2012_dm6(genetic_maps.GeneticMap):
...
-    """
+    Model Name:
+        SheehanSongThreeEpoch
+    
+    Model Description:
+        The three epoch (modern, bottleneck, ancestral) model estimated for a
+        single African Drosophila Melanogaster population from `Sheehan and Song <https:/
+        /doi.org/10.1371/journal.pcbi.1004845>`_ . Population sizes are estimated by a 
+        deep learning model trained on simulation data. NOTE: Due to differences in 
+        coalescence units between PSMC (2N) and msms (4N) generation times were doubled
```
Comment (jeromekelleher) [inline stdpopsim/drosophila_melanogaster.py]: Does this last note help? Everything is in units of generations here, so the scaling used in the original publication doesn't really enter into it, does it?

**Comment by ndukler** [comment]
Think this is good to go @jeromekelleher

**Comment by ndukler** [comment]
Actually hold off for a bit. I want to run the wording of the note on the Sheehan et al. model by @andrewkern .

**Comment by jeromekelleher** [comment]
> Actually hold off for a bit. I want to run the wording of the note on the Sheehan et al. model by @andrewkern .

Cool - merge whenever you're happy with it @andrewkern.

File: stdpopsim/drosophila_melanogaster.py [hunk 2]
```diff
@@ -77,16 +77,40 @@ class Comeron2012_dm6(genetic_maps.GeneticMap):
...
-    .. todo:: document this model, including the original publications
-        and clear information about what the different population indexes
-        mean.
+    Model Description:
+        The three epoch (modern, bottleneck, ancestral) model estimated for a
+        single African Drosophila Melanogaster population from `Sheehan and Song <https:/
+        /doi.org/10.1371/journal.pcbi.1004845>`_ . Population sizes are estimated by a
+        deep learning model trained on simulation data. NOTE: Due to differences in
+        coalescence units between PSMC (2N) and msms (4N) the number of generations were
+        doubled from PSMC estimates when simulating data from msms.
```
Comment (andrewkern) [inline stdpopsim/drosophila_melanogaster.py]: I think we should say in the note "Due to differences in
        coalescence units between PSMC (2N) and msms (4N) the number of generations were
        doubled from PSMC estimates when simulating data from msms in the original publication. We have faithfully represented the published model here"

Comment (andrewkern) [reply stdpopsim/drosophila_melanogaster.py]: I just want to make sure this doesn't sound like a bug that we have injected for no reason...

File: stdpopsim/drosophila_melanogaster.py (hunk 1 above)
Comment (andrewkern) [reply stdpopsim/drosophila_melanogaster.py]: so unfortunately the authors seem to have had their own factor of two error and we should probably document it....

File: stdpopsim/drosophila_melanogaster.py (hunk 2 above)
Comment (ndukler) [reply stdpopsim/drosophila_melanogaster.py]: I agree. I figured more experienced word-smithing was called for for this :)

**Comment by ndukler** [comment]
Ok, wording changed. @andrewkern @jeromekelleher

**Pull request merged**